### DISCUSSION TITLE
How to get the original photo capture date on photos

### DISCUSSION THREAD

**Discussion by camleng** [steilerDev/icloud-photos-sync#567] [Q&A]
The dates on each of my photos have the date that it was downloaded from the utility. Is there any way to have the original date the photo was taken to be the date on the photo?

**Comment by steilerDev** [comment]
The tool [should set the date of the picture to the "modified" date from the iCloud Photo library](https://github.com/steilerDev/icloud-photos-sync/blob/f26620dd3a8d5d66f283342656735d8ad60b3eb7/app/src/lib/icloud/icloud-photos/icloud-photos.ts#L569). Can you check which date iCloud reports on a specific asset?

**Reply by camleng** [reply]
Ok, it looks like for each of the photos in the initial run of the utility, the Modified Date is August 7, 2024. But each photo/video I've taken since then has the correct date associated with it. Is there any way to get the original date for the assets from the original run?

**Comment by alcroito** [comment]
I also have the same problem. 
A lot of my pictures have the modified date set to the date the sync was ran, rather when the picture was taken, which was months ago. 

In my case, I think this has something to do with the fact that the pictures were removed from my icloud library at some point (for yet unknown to me reasons), and I restored (re-imported) them into my library from a time machine backup.

Here is the info for a file shown as stored directly on my mac library inside `Photos Library.photoslibrary` using imagemagick 7
```
identify -verbose '/Users/user/Pictures/Photos Library.photoslibrary/originals/0/<uuid>.heic'
```

```
  Permissions: rw-------
  Format: HEIC (High Efficiency Image Format)
  Mime type: image/heic
  Class: DirectClass
  Geometry: 2316x3088+0+0
...
    date:create: 2025-02-28T21:02:17+00:00
    date:modify: 2025-01-01T16:10:00+00:00
    date:timestamp: 2025-03-02T16:00:01+00:00
    exif:DateTime: 2024:12:31 18:10:25
    exif:DateTimeDigitized: 2024:12:31 18:10:25
    exif:DateTimeOriginal: 2024:12:31 18:10:25
    photoshop:DateCreated: 2024-12-31T18:10:25
    signature: d68f50b2c35c52fae763e9c164ffe281766814f2644b703dfb81fa76b962e1be
    unknown: iPhone 13 Pro
    xmp:CreateDate: 2024-12-31T18:10:25
    xmp:CreatorTool: 18.2
    xmp:ModifyDate: 2024-12-31T18:10:25
```

And `ls -lT` reports
```
Jan  1 17:10
1073909 Jan  1 17:10:00 2025
```

So the jan 1 17:10 matches the `date:modify: 2025-01-01T16:10:00` accounting for timezone.

The picture was originally taken on an iphone on December 31st around 18:00, and likely synced to my mac via icloud the next day, on January 1st, which matches  date:modify and xmp:CreateDate.

Then I re-imported the picture from my backup on February 28th, which matches `date:create:2025-02-28`

Then i ran icloud-photos-sync on my server on the 2nd of March (today) to sync all pictures, and here is what it reports according to imagemagick:
```
  Permissions: rw-r--r--
  Format: HEIC (High Efficiency Image Format)
  Mime type: image/heic
  Class: DirectClass
  Geometry: 2316x3088+0+0
...
    date:create: 2025-03-02T11:39:02+00:00
    date:modify: 2025-02-28T21:07:33+00:00
    date:timestamp: 2025-03-02T16:09:23+00:00
    exif:DateTime: 2024:12:31 18:10:25
    exif:DateTimeDigitized: 2024:12:31 18:10:25
    exif:DateTimeOriginal: 2024:12:31 18:10:25
    photoshop:DateCreated: 2024-12-31T18:10:25
    signature: d68f50b2c35c52fae763e9c164ffe281766814f2644b703dfb81fa76b962e1be
    unknown: iPhone 13 Pro
    xmp:CreateDate: 2024-12-31T18:10:25
    xmp:CreatorTool: 18.2
    xmp:ModifyDate: 2024-12-31T18:10:25
```

and

```
ls --full-time _All-Photos/<uuid>.heic
-rw-r--r-- 1 root root 1.1M 2025-02-28 22:07:33.809000000 +0100 _All-Photos/<uuid>.heic
```

So while the exif and xmp dates stayed the same, date:modify changed to `2025-02-28T21:07:33`, which somewhat matches, but not exactly the previous date:create.

Accessing the picture from the icloud photos web interface via browser still reports 'Tuesday · Dec 31, 2024 · 6:10 PM' though.

I'm guessing the file as stored in icloud had a changed modify time, and that affects the sync here.

I think it would be nice to provide an option that could set the modified time either to the exif date time, the photoshop one, or the xmp one.
